### PR TITLE
:bar_chart: Create Colonial Dates Dataset (COLDAT) (rebased)

### PR DESCRIPTION
**Pull request by Marigold**
Rebased https://github.com/owid/etl/pull/1614

Uploads COLDAT to the ETL, as part of the [State Capacity topic page](https://github.com/owid/owid-issues/issues/925).

Staging version of the dataset here: http://staging-site-parriagadap/admin/datasets/6218

### CODE REVIEW

File: etl/steps/data/garden/harvard/2023-09-18/colonial_dates_dataset.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    # Create an additional summarized colonizer column, replacing the values with " - " with "More than one colonizer"
+    # I add the "z." to have this at the last position of the map brackets
+    tb["colonizer_grouped"] = tb["colonizer"].apply(
+        lambda x: "z. Multiple colonizers" if isinstance(x, str) and " - " in x else x
+    )
+
+    # Create last_colonizer column, which is the most recent non-null colonizer for each country and year
+    tb["last_colonizer"] = tb.groupby(["country"])["colonizer"].fillna(method="ffill")
+    tb["last_colonizer_grouped"] = tb.groupby(["country"])["colonizer_grouped"].fillna(method="ffill")
```
Comment (paarriagadap) [inline etl/steps/data/garden/harvard/2023-09-18/colonial_dates_dataset.py:86]: I see metadata is not kept when using this and not what it is commented, @Marigold. Am I doing something wrong?

Comment (Marigold) [reply etl/steps/data/garden/harvard/2023-09-18/colonial_dates_dataset.py:86]: Thanks for spotting this! It turned out that `fillna` behaves differently from other groupby functions. I pushed fix to master and rebased this branch (you can get the changes with `git fetch --all && git reset --hard origin/fix-create-colonial-dates-dataset`)

File: tests/backport/datasync/test_data_metadata.py
Comment (paarriagadap) [inline tests/backport/datasync/test_data_metadata.py:1]: @Marigold In this rebased PR there are some files not from the original PR. Will it be OK to merge it anyway?

Comment (Marigold) [reply tests/backport/datasync/test_data_metadata.py:1]: Yeah, not sure how they got in (maybe some automatic whitespace correction?), but it's safe to merge.

**Pull request merged**